airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt

```diff
@@ -373,46 +373,30 @@ internal constructor(
 
         private lateinit var validator: JsonSchemaValidator
 
-        @VisibleForTesting
-        @Throws(Exception::class)
-        fun consumeWriteStream(consumer: SerializedAirbyteMessageConsumer) {
-            BufferedInputStream(System.`in`).use { bis ->
-                ByteArrayOutputStream().use { baos -> consumeWriteStream(consumer, bis, baos) }
-            }
-        }
-
-        @VisibleForTesting
         @Throws(Exception::class)
-        fun consumeWriteStream(
+        internal fun consumeWriteStream(
             consumer: SerializedAirbyteMessageConsumer,
-            bis: BufferedInputStream,
-            baos: ByteArrayOutputStream
+            inputStream: BufferedInputStream = BufferedInputStream(System.`in`)
         ) {
-            consumer.start()
-
-            val buffer = ByteArray(8192) // 8K buffer
-            var bytesRead: Int
-            var lastWasNewLine = false
-
-            while ((bis.read(buffer).also { bytesRead = it }) != -1) {
-                for (i in 0 until bytesRead) {
-                    val b = buffer[i]
-                    if (b == '\n'.code.toByte() || b == '\r'.code.toByte()) {
-                        if (!lastWasNewLine && baos.size() > 0) {
-                            consumer.accept(baos.toString(StandardCharsets.UTF_8), baos.size())
-                            baos.reset()
-                        }
-                        lastWasNewLine = true
+            inputStream.bufferedReader(StandardCharsets.UTF_8).use {
+                var emptyLines = 0
+                it.forEachLine { line: String ->
+                    if (line.isNotEmpty()) {
+                        consumer.accept(line, line.toByteArray(StandardCharsets.UTF_8).size)
                     } else {
-                        baos.write(b.toInt())
-                        lastWasNewLine = false
+                        emptyLines++
+                        // We've occasionally seen this loop not exit
```

Comment:
we're not adding any loop, so I don't think we're making the problem worse. And once every 1k lines should not be a problem. If it is, it's a problem on the side that's generating those empty lines IMO